src/App.tsx

```diff
@@ -1,27 +1,106 @@
-import { render } from '@testing-library/react';
-import React from 'react';
-import './App.css';
-import Contacts from './Contacts';
-import Header from './Header';
-import Sidebar from './Sidebar';
-import Feed from './Feed';
+import React, { useEffect, useState } from "react";
+import "./App.css";
+import Contacts from "./Contacts";
+import Header from "./Header";
+import Sidebar from "./Sidebar";
+import Feed from "./Feed";
+import CreatePostWindow from "./CreatePostWindow";
 
 function App() {
+  let username = "Meir";
+
+
+  
+  const [postData, setPostData] = useState([""]);
+  const addPost = (input: string) => {
+    setPostData([input, ...postData]);
+  };
+
+  const [isNewPostWindowAlive, setWindowState] = useState(false);
+  const toggleWindow = () => {
+    if (isNewPostWindowAlive && isNewPostInProcess) setGlobalTemp(""); //dispach input from createPostWindow to MessageSenderPlaceholder
+    console.log("clicked");
+    setWindowState((isNewPostWindowAlive) => !isNewPostWindowAlive);
+    if (!isNewPostInProcess) setGlobalTemp(`What's on your mind, ${username}?`);
+  };
+  // update: A->D && P->P input
+  // update: A->D && P->I placeholder
+
+  const [isNewPostInProcess, setNewPostInProcess] = useState(false);
+  const newPostInProcess = (b: boolean) => {
+    console.log(isNewPostWindowAlive);
+    // if (!b && !isNewPostWindowAlive) setGlobalTemp(`What's onmindxxx, ${username}?`);
+    // if (b) document.getElementsByClassName("create_post__button")[0].setAttribute('style', 'background-color:#2374E1; color:white; cursor:pointer' );
+    // if (!b) document.getElementsByClassName("create_post__button")[0].removeAttribute('style');
+    setNewPostInProcess(b);
+  };
+
+  const [globalTempNewPost, setGlobalTempNewPost] = useState(
+    `What's on your mind, ${username}?`
+  );
+
+  const setGlobalTemp = async (input: string) => {
+    if (input.replace(/\s/g, "") === "") setGlobalTempNewPost(localTempNewPost);
+    else setGlobalTempNewPost(input);
+  };
+
+  const [localTempNewPost, setLocalTempNewPost] = useState("");
+  const setLocalTemp = (input: string) => {
+    setLocalTempNewPost(input);
+  };
+
+  
+  useEffect(() => {
+    if (!isNewPostInProcess && !isNewPostWindowAlive)
+      setGlobalTemp(`What's on your mind, ${username}?`);
+  });
   return (
     <div className="App">
-      <div className="Header"><Header /></div>
+      <div
+        className="opacitySolid"
+        onClick={() => {
+          document
+            .getElementsByClassName("opacitySolid")[0]
```

Comment:
try not to use this type of function with react, if you want to change the style you can change the class according to a state: `className={itsClassX? classXName : classYName}`

Reply:
Is it the sole way to do it React-style?
What about changing only the content of "style", e.g. 

```
 const button_styles = {
    backgroundColor: `${props.Temp!== "" ? "#2374E1" : "#505151" }`,
  }; 

<button style={button_styles}>
```

Reply:
There are many tactics to deal with style in react, this is a good point to start:
https://blog.logrocket.com/styling-in-react-4-ways-style-react-app/

Here you can see you can handle styles in a js file, with that you can pass to it parameters and make it dynamic.

I saw what you did with the style with js, it's a good idea, but rest the article to get an idea on how to organize this better.

But never with operation directly on Dom

Reply:
My main question is, what is the proper way to change style according to state?
I understand there is no one way, but as long as it is an understandable one and not messing with DOM - it's fine.
Perhaps I'm wrong and the best way is changing className?

Reply:
How do I change `body` styles?
for example: `document.body.style.overflow = "hidden"
`
(in order the make the scrolling bar disappear)